### PR TITLE
Clean up IP header dependencies in TCP2

### PR DESCRIPTION
**Pull request by pfl**
Use existing functions to compute the start of the TCP header. Add a check on CONFIG_HEAP_MEM_POOL_SIZE in order to have memory for k_{c,m}alloc used by tcp2.c.

DNM until tested.

### CODE REVIEW

File: subsys/net/ip/Kconfig [hunk 1]
```diff
@@ -371,10 +371,12 @@ config NET_TCP1
 	help
 	  The current TCP stack that has been in use since Zephyr 1.0.
 
+if HEAP_MEM_POOL_SIZE > 0
 config NET_TCP2
 	bool "Experimental TCP stack"
 	help
```
Comment (jukkar) [inline subsys/net/ip/Kconfig]: We could do this also by `depends on HEAP_MEM_POOL_SIZE > 0`

**Comment by pfl** [comment]
Some more IPv4/IPv6 dependencies remain, will make an attempt on them during the next few days.

File: samples/net/sockets/echo_client/prj.conf [hunk 2]
```diff
@@ -2,13 +2,16 @@
 CONFIG_NETWORKING=y
 CONFIG_NET_UDP=y
 CONFIG_NET_TCP=y
+#CONFIG_NET_TCP2=y
```
Comment (jukkar) [inline samples/net/sockets/echo_client/prj.conf]: Please create an overlay config file `overlay-tcp2.conf` for these settings.

**Review by jukkar — CHANGES_REQUESTED** [review]
Looks good, couple comments only.

**Review by ozhuraki — CHANGES_REQUESTED** [review]
@pfl

I suggest to split this pull request into 2:

-	Handling the heap. I think the proper way is to convert it into slab
-	Header access

For the header access, I would like the change to be postponed after we add IPv6 support.

**Comment by pfl** [comment]
>     * Handling the heap. I think the proper way is to convert it into slab

This part should be done as in net_tcp_alloc() in the current tcp.c TCP stack.

**Review by tbursztyka — COMMENTED** [review]
besides jukka's comments, lgtm.

File: samples/net/sockets/echo_client/prj.conf (hunk 2 above)
Comment (pfl) [reply samples/net/sockets/echo_client/prj.conf]: Done.

**Pull request closed without merging**